cmd/docker/docker.go

```diff
@@ -312,63 +312,69 @@ type versionDetails interface {
 	ServerInfo() command.ServerInfo
 }
 
-func hideFeatureFlag(f *pflag.Flag, hasFeature bool, annotation string) {
-	if hasFeature {
+func hideFeatureFlag(f *pflag.Flag, hasFeature func(string) bool, annotation string) {
+	if f.Hidden {
 		return
 	}
-	if _, ok := f.Annotations[annotation]; ok {
-		f.Hidden = true
+	if values, ok := f.Annotations[annotation]; ok {
+		var v string
+		if len(values) > 0 {
+			v = values[0]
+		}
+		if !hasFeature(v) {
+			f.Hidden = true
+		}
 	}
 }
 
-func hideFeatureSubCommand(subcmd *cobra.Command, hasFeature bool, annotation string) {
-	if hasFeature {
+func hideFeatureSubCommand(subcmd *cobra.Command, hasFeature func(string) bool, annotation string) {
+	if subcmd.Hidden {
 		return
 	}
-	if _, ok := subcmd.Annotations[annotation]; ok {
-		subcmd.Hidden = true
+	if v, ok := subcmd.Annotations[annotation]; ok {
+		if !hasFeature(v) {
+			subcmd.Hidden = true
+		}
 	}
 }
 
 func hideUnsupportedFeatures(cmd *cobra.Command, details versionDetails) error {
-	clientVersion := details.Client().ClientVersion()
-	osType := details.ServerInfo().OSType
-	hasExperimental := details.ServerInfo().HasExperimental
-	hasExperimentalCLI := details.ClientInfo().HasExperimental
-	hasBuildKit, err := command.BuildKitEnabled(details.ServerInfo())
-	if err != nil {
-		return err
-	}
+	var (
+		hasClientVersion   = func(v string) bool { return versions.LessThan(details.Client().ClientVersion(), v) }
```

Comment:
Shouldn't we use `LessThanOrEqualTo` instead?

Reply:
hm, yes it should be reversed here; we need to hide if lower than the version in the annotation.

Let me also rename these versions and reverse the logic because we're now doing `if !hasFeature(v) {`, which is slightly confusing